pavement.py

```diff
@@ -562,25 +558,21 @@ def sdist(options):
         target = os.path.join(idirs, tarball_name(t))
         shutil.copy(source, target)
 
-def compute_md5(idirs):
+def _compute_hash(idirs, algo):
     released = paver.path.path(idirs).listdir()
     checksums = []
     for f in sorted(released):
-        m = md5(open(f, 'r').read())
+        m = algo(open(f, 'r').read())
```

Comment:
While you're here, mind making a second commit that uses `with` to correctly close the file?